pkg/apis/agent/v1alpha1/agent_types.go

```diff
@@ -0,0 +1,223 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package v1alpha1
+
+import (
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	commonv1 "github.com/elastic/cloud-on-k8s/pkg/apis/common/v1"
+)
+
+// AgentSpec defines the desired state of the Agent
+type AgentSpec struct {
+	// Version of the Agent.
+	Version string `json:"version"`
+
+	// ElasticsearchRefs is a reference to a list of Elasticsearch clusters running in the same Kubernetes cluster.
+	// Due to existing limitations, only a single ES cluster is currently supported.
```

Comment:
Are you planning on adding a validation check to enforce this?

Reply:
Yes, as part of https://github.com/elastic/cloud-on-k8s/issues/4012. As to the `elasticsearchRefs`, I'd like to take a long shot at supporting it on the initial release, already chatted with @barkbay about this a bit.